### PR TITLE
Check Nova AZ and additional networks

### PR DESCRIPTION
**Pull request by itzikb-redhat**
Make sure the Nova AZ and additional networks in machines are the same as Openstack instances

### CODE REVIEW

File: test/extended/openstack/machine.go [hunk 1]
```diff
@@ -95,10 +119,14 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] Machine", func()
 			})
 
 			g.By(fmt.Sprintf("Comparing Machine %q with instance %q: root volume", machineName, instance.Name), func() {
-				if instance.Image["id"] != nil {
-					instanceImage, err := images.Get(computeClient, fmt.Sprintf("%v", instance.Image["id"])).Extract()
-					o.Expect(err).NotTo(o.HaveOccurred())
-					o.Expect(instanceImage.Name).To(o.Equal(machineImage), "Image not matching for instance %q", instance.Name)
+				for _, vol := range instance.AttachedVolumes {
+					v, err := volumes.Get(computeClient, vol.ID).Extract()
```
Comment (MaysaMacedo) [inline test/extended/openstack/machine.go:123]: This is failing with:
```
fail [github.com/openshift/openstack-test/test/extended/openstack/machine.go:124]: Error fetching Openstack volumes for Machine "k0r4ltht-a50fa-slrwh-worker-0-ksw9k"
Unexpected error:
    <gophercloud.ErrDefault404>: 
    Resource not found: [GET https://compute.rdo.mtl2.vexxhost.net/v2.1/volumes/f35ae143-8c2d-4507-8c6b-74729ca6b550], error message: {"message": "The resource could not be found.<br /><br />\n\n\n", "code": "404 Not Found", "title": "Not Found"}
```
maybe a timeout?

File: test/extended/openstack/machine.go [hunk 2]
```diff
@@ -64,18 +70,36 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] Machine", func()
...
+			var machineNetwork *networks.Network
 			machineName := machine.Get("metadata.name").String()
 			machineFlavor := machine.Get("spec.providerSpec.value.flavor").String()
 			machineImage := machine.Get("spec.providerSpec.value.image").String()
+			machineAZ := machine.Get("spec.providerSpec.value.availabilityZone").String()
+			machineRootVolCinderAz := machine.Get("spec.providerSpec.value.rootVolume.availabilityZone").String()
+			machineRootVolVolumeType := machine.Get("spec.providerSpec.value.rootVolume.volumeType").String()
 			machineSecurityGroups := make(map[string]struct{})
+			for _, network := range objects(machine.Get("spec.providerSpec.value.networks")) {
+				if network.Get("subnets").String() == "" {
```
Comment (MaysaMacedo) [inline test/extended/openstack/machine.go:90]: why the machinenetwork name is only being retrieved when subnet is not defined?

Comment (itzikb-redhat) [reply test/extended/openstack/machine.go:90]: To get only the additional networks

File: test/extended/openstack/machine.go (hunk 1 above)
Comment (itzikb-redhat) [reply test/extended/openstack/machine.go:123]: I Saw what and I wondered if there was  a problem with the code

File: test/extended/openstack/machine.go [hunk 3]
```diff
@@ -95,10 +122,14 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] Machine", func()
 			})
 
 			g.By(fmt.Sprintf("Comparing Machine %q with instance %q: root volume", machineName, instance.Name), func() {
-				if instance.Image["id"] != nil {
-					instanceImage, err := images.Get(computeClient, fmt.Sprintf("%v", instance.Image["id"])).Extract()
-					o.Expect(err).NotTo(o.HaveOccurred())
-					o.Expect(instanceImage.Name).To(o.Equal(machineImage), "Image not matching for instance %q", instance.Name)
```
Comment (pierreprinetti) [inline test/extended/openstack/machine.go:101]: Why is the image test going away?

Comment (itzikb-redhat) [reply test/extended/openstack/machine.go:101]: It's not. It as duplicated.

File: test/extended/openstack/machine.go [hunk 4]
```diff
@@ -64,18 +73,36 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] Machine", func()
...
 			machineName := machine.Get("metadata.name").String()
 			machineFlavor := machine.Get("spec.providerSpec.value.flavor").String()
 			machineImage := machine.Get("spec.providerSpec.value.image").String()
+			machineAZ := machine.Get("spec.providerSpec.value.availabilityZone").String()
+			machineRootVolCinderAz := machine.Get("spec.providerSpec.value.rootVolume.availabilityZone").String()
+			machineRootVolVolumeType := machine.Get("spec.providerSpec.value.rootVolume.volumeType").String()
 			machineSecurityGroups := make(map[string]struct{})
+			for _, network := range objects(machine.Get("spec.providerSpec.value.networks")) {
+				if network.Get("subnets").String() == "" {
+					machineNetwork, err = networks.Get(networkClient, network.Get("uuid").String()).Extract()
```
Comment (MaysaMacedo) [inline test/extended/openstack/machine.go:94]: the err returned needs to be handled

**Review by MaysaMacedo — COMMENTED** [review]
Just one remark and after that it lgtm

**Pull request merged**